### PR TITLE
119 Fix-auth-callback-url-for-flutter-app

### LINKED ISSUE #119: Fix Auth Callback url for Flutter app
**Short description:**

All auth operation doesn't work on mobile app.

**Describe the solution you'd like**
The current problem is caused by the callback url.
Maybe the setup of an activity can handle this problem.

### PR DESCRIPTION
**Pull request by LucasTesnier**
# Purpose

The purpose of this pull request is to fix the Auth callbackUrl bug for Android version.

Fixes #119 

## Fixed feature

* A new Deep Link activity have been set for the url : https://www.test.com (It will be updated after the passage to production #104)
* A Native code have been settled for Web or Android which decide to the correct callbackUrl to use.

### CODE REVIEW

File: application/app_code/lib/pages/auth_linker/auth_linker_functional.dart [hunk 1]
```diff
@@ -153,19 +154,21 @@ Future<String> getDiscordToken() async {
     final url = Uri.https('discord.com', '/api/oauth2/authorize', {
       'response_type': 'code',
       'client_id': clientId,
-      'redirect_uri': 'http://localhost:8081/auth.html',
+      'redirect_uri':
+          !kIsWeb ? 'https://www.test.com' : 'http://localhost:8081/auth.html',
```
Comment (ZQUEMA) [inline application/app_code/lib/pages/auth_linker/auth_linker_functional.dart:158]: Do you realy want to keep 'https://www.test.com' ?

File: application/app_code/lib/pages/auth_linker/auth_linker_functional.dart [hunk 2]
```diff
@@ -153,19 +154,21 @@ Future<String> getDiscordToken() async {
...
+        url: url.toString(), callbackUrlScheme: !kIsWeb ? 'https' : 'http');
 
     final code = Uri.parse(result).queryParameters['code'];
 
     final response = await http
         .post(Uri.parse('https://discord.com/api/oauth2/token'), body: {
       'client_id': clientId,
-      'redirect_uri': 'http://localhost:8081/auth.html',
+      'redirect_uri':
+          !kIsWeb ? 'https://www.test.com' : 'http://localhost:8081/auth.html',
```
Comment (ZQUEMA) [inline application/app_code/lib/pages/auth_linker/auth_linker_functional.dart:171]: Same as above

File: application/app_code/lib/pages/auth_linker/auth_linker_functional.dart [hunk 3]
```diff
@@ -190,9 +193,7 @@ Future<String> getDiscordToken() async {
 
 /// Invite the Discord bot to the user server
 Future<String> inviteDiscordBot() async {
-  await launchUrl(
-      Uri.parse(
-          'https://discord.com/api/oauth2/authorize?client_id=1066384923231006741&permissions=8&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2F&response_type=code&scope=bot%20identify%20guilds%20email%20connections%20messages.read%20guilds.members.read%20guilds.join'),
+  await launchUrl(Uri.parse('https://www.test.com'),
```
Comment (ZQUEMA) [inline application/app_code/lib/pages/auth_linker/auth_linker_functional.dart:196]: same as above

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
Answer the conversation

**Review by TomDUVAL-MAHE — APPROVED** [review]
Other that comments made by Quentin, NTR

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**